soc/arm/nordic_nrf/nrf54h/soc.c

```diff
@@ -53,23 +57,21 @@ static void power_domain_init(void)
 #endif
 }
 
-#if defined(CONFIG_SOC_TRIM_HSFLL)
+#if HSFLL_TRIM
 static int trim_hsfll(void)
 {
 	NRF_HSFLL_Type *hsfll = (NRF_HSFLL_Type *)DT_REG_ADDR(HSFLL_NODE);
-	const nrf_hsfll_trim_t trim = {
-		.vsup = FICR_TRIM_BLOCK.HSFLL.TRIM.VSUP,
-		.coarse = FICR_TRIM_BLOCK.HSFLL.TRIM.COARSE[CONFIG_FICR_HSFLL_TRIM_IDX],
-		.fine = FICR_TRIM_BLOCK.HSFLL.TRIM.FINE[CONFIG_FICR_HSFLL_TRIM_IDX],
-	};
+	nrf_hsfll_trim_t trim;
+
+	trim.vsup = sys_read32(FICR_ADDR_GET(HSFLL_NODE, vsup));
+	trim.coarse = sys_read32(FICR_ADDR_GET(HSFLL_NODE, coarse));
+	trim.fine = sys_read32(FICR_ADDR_GET(HSFLL_NODE, fine));
 
 	LOG_DBG("Trim: HSFLL VSUP: 0x%.8x", trim.vsup);
 	LOG_DBG("Trim: HSFLL COARSE: 0x%.8x", trim.coarse);
 	LOG_DBG("Trim: HSFLL FINE: 0x%.8x", trim.fine);
 
-	if (CONFIG_HSFLL_CLOCK_MULTIPLIER) {
-		nrf_hsfll_clkctrl_mult_set(hsfll, CONFIG_HSFLL_CLOCK_MULTIPLIER);
-	}
+	nrf_hsfll_clkctrl_mult_set(hsfll, DT_PROP(HSFLL_NODE, multiplier));
```

Comment:
note here: multiplier is only applicable when working in closed loop mode, however, I assume configuring it to the right value won't hurt. This code snipped doesn't seem to configure the operating mode.